server/service/src/requisition_line/response_requisition_line/insert/validate.rs

```diff
@@ -0,0 +1,50 @@
+use repository::{RequisitionRow, RequisitionStatus, RequisitionType, StorageConnection};
+
+use crate::{
+    item::check_item_exists,
+    requisition::common::check_requisition_row_exists,
+    requisition_line::common::{check_item_exists_in_requisition, check_requisition_line_exists},
+};
+
+use super::{InsertResponseRequisitionLine, OutError};
+
+pub fn validate(
+    connection: &StorageConnection,
+    store_id: &str,
+    input: &InsertResponseRequisitionLine,
+) -> Result<RequisitionRow, OutError> {
+    if (check_requisition_line_exists(connection, &input.id)?).is_some() {
+        return Err(OutError::RequisitionLineAlreadyExists);
+    }
+
+    let requisition_row = check_requisition_row_exists(connection, &input.requisition_id)?
+        .ok_or(OutError::RequisitionDoesNotExist)?;
+
+    if requisition_row.program_id.is_some() {
+        return Err(OutError::CannotAddItemToProgramRequisition);
+    }
+
+    if requisition_row.store_id != store_id {
+        return Err(OutError::NotThisStoreRequisition);
+    }
+
+    if requisition_row.status != RequisitionStatus::New {
```

Comment:
I suppose `Draft` should be allowed too? But doesn't really matter because they start at `New` anyway. Wondering if we even need draft lol.

Reply:
There shouldn't be any response requisitions with Draft 🤔 and frontend doesn't allow editing of Draft response requisitions, so frontend will need changes to if we account for it (even though they shouldn't be able to create draft response requisitions

Reply:
Yeah no I totally agree!